packages/synapse-react-client/src/components/ChangePassword/useChangePasswordFormState.tsx

```diff
@@ -6,16 +6,23 @@ import {
   PasswordResetSignedToken,
   TwoFactorAuthErrorResponse,
   TwoFactorAuthOtpType,
+  TwoFactorAuthResetRequest,
 } from '@sage-bionetworks/synapse-types'
-import { useChangePassword } from '../../synapse-queries'
+import { useChangePassword, useResetTwoFactorAuth } from '../../synapse-queries'
 import { Alert, Typography } from '@mui/material'
 import {
   RECOVERY_CODE_GUIDANCE_TEXT_SHORT,
   TOTP_GUIDANCE_TEXT,
 } from '../Authentication/Constants'
 import { ONE_TIME_PASSWORD_STEP, OneTimePasswordForm } from '../Authentication'
+import { useOneSageURL } from '../../utils/hooks'
+import appendFinalQueryParamKey from '../../utils/appendFinalQueryParamKey'
+
+export const TWO_FACTOR_AUTH_CHANGE_PASSWORD_PROMPT =
+  'Two-factor authentication is required to change your password. Your password has not yet been changed.'
 
 export type UseChangePasswordFormStateOptions = {
+  hideReset2FA?: boolean
```

Comment:
If changing the password on the AccountSettings page (user is logged in), we will hide the reset 2FA button. The user can just disable 2FA using the button on the page